pkg/batcher/gate.go

```
@@ -0,0 +1,132 @@
+// than sending, so it never blocks and every waiter observes it.
+func (g *admissionGate) signalEmpty() {
+	g.emptyOnce.Do(func() { close(g.emptyCh) })
+}
+
+// empty returns the latch that closes once no publisher remains in the window.
+func (g *admissionGate) empty() <-chan struct{} {
+	return g.emptyCh
+}
+
+// sealed reports whether admission is closed.
+func (g *admissionGate) isSealed() bool {
```

golangci-lint (2.12.2)

[error] func (*admissionGate).isSealed is unused

(unused)